packages/ui/src/components/editor/panels/Files/browserGrid/index.tsx

```diff
@@ -416,21 +411,31 @@ export function FileBrowserItem({
         <Button variant="outline" size="small" fullWidth onClick={viewAssetProperties}>
           {t('editor:layout.filebrowser.viewAssetProperties')}
         </Button>
-        <Button
-          variant="outline"
-          size="small"
-          fullWidth
-          disabled={!fileConsistsOfContentType(item, 'model')}
-          onClick={() => {
-            if (fileConsistsOfContentType(item, 'model')) {
-              PopoverState.showPopupover(
-                <ModelCompressionPanel selectedFile={item as FileType} refreshDirectory={refreshDirectory} />
-              )
-            }
-          }}
-        >
-          {t('editor:layout.filebrowser.compress')}
-        </Button>
+
+        {fileConsistsOfContentType(item, 'model') && (
+          <Button
+            variant="outline"
+            size="small"
+            fullWidth
+            // disabled={!fileConsistsOfContentType(item, 'model')} // TODO: move context menu to its own component, with a State<Filetype[]> -JS
+            onClick={(e) => openModelCompress()}
+          >
+            {t('editor:layout.filebrowser.compress')}
+          </Button>
+        )}
+
+        {fileConsistsOfContentType(item, 'image') && (
+          <Button
+            variant="outline"
+            size="small"
+            fullWidth
+            // disabled={!fileConsistsOfContentType(item, 'image')} // TODO: move context menu to its own component, with a State<Filetype[]> -JS
+            onClick={(e) => setOpenImageCompress(true)}
+          >
+            {t('editor:layout.filebrowser.compress')}
+          </Button>
+        )}
```

Comment:
Agreed; my change here is an intentional stopgap that allows the context menus to surface the proper compression panel (image vs model) without requiring knowledge of every selected file.

Once the context menu is moved to the container tsx, we'll be able to modify its state to be an array of files instead of just one item, and the previous code can triumphantly return 🌟

Reply:
new compression and files properties modals now in file browser container